sway and wlc versions are not compatible

## Issue description

When trying to build sway, it fails because wlc changed the API (https://github.com/Cloudef/wlc/commit/e9d8e985202ecd3e6219252b75553204b41541c5) that sway relies on.

### Steps to reproduce

Install sway from the nixpkgs master. It will fail with the following error message:
```/tmp/nix-build-sway-0.8.drv-0/sway-0.8-src/sway/config.c:868:3: error: too few arguments to function 'wlc_output_set_resolution'
   wlc_output_set_resolution(output->handle, &new_size);```

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
  - nixos-16.09pre90254.6b20d5b
* Nix version: (run `nix-env --version`)
  - nix-env (Nix) 1.11.3
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
  - 17.03.git.d408070

--- a/pkgs/applications/window-managers/sway/default.nix
+++ b/pkgs/applications/window-managers/sway/default.nix
@@ -4,7 +4,7 @@
 }:
 
 let
-  version = "0.8";
+  version = "0.9";
 in
   stdenv.mkDerivation rec {
     name = "sway-${version}";
@@ -13,7 +13,7 @@ in
       owner = "Sircmpwn";
       repo = "sway";
       rev = "${version}";
-      sha256 = "10i62cn1z7fwg0jwkskmzcaha39lprkl4zvkp59jr5wvpjligdq3";
+      sha256 = "0qqqg23rknxnjcgvkfrx3pijqc3dvi74qmmavq07vy2qfs1xlwg0";
     };
 
     nativeBuildInputs = [ makeWrapper cmake pkgconfig asciidoc libxslt docbook_xsl ];

